Kconfig.zephyr

```diff
@@ -403,6 +403,19 @@ config STD_C23
 
 endchoice
 
+config TOOLCHAIN_SUPPORTS_GNU_EXTENSIONS
+	bool
+	default y if "$(ZEPHYR_TOOLCHAIN_VARIANT)" = "zephyr" || "$(ZEPHYR_TOOLCHAIN_SUPPORTS_GNU_EXTENSIONS)" = "y"
```

Comment:
what is: `ZEPHYR_TOOLCHAIN_SUPPORTS_GNU_EXTENSIONS` ?

Also we should have an update here: https://github.com/zephyrproject-rtos/sdk-ng/blob/main/cmake/zephyr/Kconfig:
```
config TOOLCHAIN_ZEPHYR_SUPPORTS_GNU_EXTENSIONS
	def_bool y
	select TOOLCHAIN_SUPPORTS_GNU_EXTENSIONS
```
but the `default y if "$(ZEPHYR_TOOLCHAIN_VARIANT)" = "zephyr"` can stay as older Zephyr Toolchains also support gnu extensions.

Reply:
I did the same as was done here:
https://github.com/zephyrproject-rtos/zephyr/blob/5919bd988b9dfb40a6b4ab5a3fd4af2dccda3c95/kernel/Kconfig#L1009-L1014

Reply:
I've created a PR: https://github.com/zephyrproject-rtos/sdk-ng/pull/746

Reply:
> I did the same as was done here:

which seems to have been introduced at a point when I was away 😄 
https://github.com/zephyrproject-rtos/zephyr/pull/45656

You already added the Kconfig to indicate when cross-compile supports gnu extension or not.
And we should be careful not to introduce too many ways to achieve the same (leads to confusion), but in this particular case, I acknowledge that especially on Linux similar, setting this as env is convenient,  but the caveat is that users then forget to clear if switching to a toolchain not supporting such extensions.

So when picking up from environment, then we should only adjust default for the cross-compile, as to not impact all toolchains. Seems I have some followup to do on the `TLS` part.

But I see the setting has been removed, which I think is good for time being.